pkgs/unified_analytics/lib/src/asserts.dart

```diff
@@ -0,0 +1,126 @@
+/// Checks that the body of the request being sent to
+/// GA4 is within the limitations
+///
+/// Limitations can be found:
+/// https://developers.google.com/analytics/devguides/collection/protocol/ga4/sending-events?client_type=gtag#limitations
+
+void checkBody(Map<String, Object?> body) {
+  // Ensure we have the correct top level keys
+  if (!body.keys.contains('client_id')) {
+    throw AnalyticsException('client_id missing from top level keys');
+  }
+  if (!body.keys.contains('events')) {
+    throw AnalyticsException('events missing from top level keys');
+  }
+  if (!body.keys.contains('user_properties')) {
+    throw AnalyticsException('user_properties missing from top level keys');
+  }
+
+  final List events = body['events'] as List;
+  final Map<String, Object?> userProperties =
+      body['user_properties'] as Map<String, Object?>;
+
+  // GA4 Limitation:
+  // Requests can have a maximum of 25 events
+  if (events.length > 25) {
+    throw AnalyticsException('25 is the max number of events');
+  }
+
+  // Checks for each event object
+  for (Map<String, Object?> eventMap in events) {
+    // GA4 Limitation:
+    // Event names must be 40 characters or fewer, may only contain
+    // alpha-numeric characters and underscores, and must start
+    // with an alphabetic character
+    if ((eventMap['name'] as String).length > 40) {
```

Comment:
Consider extracting `eventMap['name'] as String` into a local variable to improve performance. While this code is only expected to be run in testing and debugging, the longer it takes to run the more chance there is that it will change the timing of other parts of the system. Also, the faster tests can run the better.